app/models/user.rb

```diff
@@ -342,14 +342,29 @@ def destroy
     super
   end
 
-  %w[veteran_status military_information payment].each do |emis_method|
-    define_method(emis_method) do
-      emis_model = instance_variable_get(:"@#{emis_method}")
-      return emis_model if emis_model.present?
-
-      emis_model = "EMISRedis::#{emis_method.camelize}".constantize.for_user(self)
-      instance_variable_set(:"@#{emis_method}", emis_model)
-      emis_model
+  military_information_method = 'military_information'
+  define_method(military_information_method) do
+    model = instance_variable_get(:"@#{military_information_method}")
+    return model if model.present?
+
+    form_profile = FormProfile.new(form_id: nil, user: self)
+    model = form_profile.initialize_military_information
+    instance_variable_set(:"@#{military_information_method}", model)
+    model
+  end
+
+  def veteran_status
+    if Flipper.enabled?(:veteran_status_updated)
```

Comment:
Any reason we have to have this flipper? Maybe a pr to introduce the `VAProfile::VeteranStatus::Service.new(self)`, test it on a rails instance or whatever, and then a PR to just swap this over? (Gotta be honest, I'm not a fan of Flipper on the backend at all)

Reply:
@LindseySaari How do you feel about removing Flipper and testing via Trevor's suggestions?

Reply:
@bosawt Can you please explain in further detail what testing the updated method on a rails instance and then swapping over entails? 

The thinking behind feature toggles was to have the feature sit in lower envs for a while and if we merge it, it'll go to prod without the FT.

cc: @LindseySaari

Reply:
@bosawt This came up once more during standup. Our team is wondering the drawbacks of Flipper on the backend.

Reply:
Oh yeah, sorry I meant to respond sooner. 

So basically I don't like `Flipper` because it is technically doubling the code flows, and teams rarely test both paths, especially ongoing maintenance. The idea of `Flipper` is that you're able to turn it on or off at any time, but in order for that to be true, you have to prove that both work. Instead what ends up happening, is you turn it 'on', and the unused code path decays and prevents you from turning it 'off'. So unless you find an immediate issue with the change, your `Flipper` had no purpose because you can't turn it off. In my opinion a revert commit is much more effective for that.

As a reviewer, if you have a `Flipper` feature, I would want you to prove that `on` and `off` both work. And if you were able to prove to me that `on` worked, what's the purpose of `off`?

Reply:
That being said, there are various reasons people may want a `Flipper` feature on the backend, and these reasons sometimes do make sense, so if you really need it then you really need it. So my main question is just to ask: 'are you really sure you really need it?'